src/PureScript/Backend/Chez/Syntax.purs

```diff
@@ -98,6 +99,25 @@ data ChezExpr
   | Identifier Prim.String
   | List (Prim.Array ChezExpr)
 
+definitionIdentifiers :: ChezDefinition -> Prim.Array Prim.String
+definitionIdentifiers (DefineValue i _) = [ i ]
+definitionIdentifiers (DefineCurriedFunction i _ _) = [ i ]
+definitionIdentifiers (DefineUncurriedFunction i _ _) = [ i ]
+definitionIdentifiers (DefineRecordType i []) =
+  [ recordTypeUncurriedConstructor i
+  , recordTypePredicate i
+  ]
+definitionIdentifiers (DefineRecordType i fields) =
+  [ recordTypeUncurriedConstructor i
+  , recordTypePredicate i
+  ] <> map (recordTypeAccessor i) fields
```

Comment:
I think matching against an empty array is redundant here? Unless there's some extra code you want to add when it's empty. (Resubmitted this comment because somehow, I might've commented on an earlier commit)
```suggestion
definitionIdentifiers (DefineRecordType i []) =
  [ recordTypeUncurriedConstructor i
  , recordTypePredicate i
  ]
definitionIdentifiers (DefineRecordType i fields) =
  [ recordTypeUncurriedConstructor i
  , recordTypePredicate i
  ] <> map (recordTypeAccessor i) fields
```